phoenix-core/src/it/java/org/apache/phoenix/end2end/InListIT.java

```diff
@@ -537,4 +539,267 @@ public void testInListExpressionWithDesc() throws Exception {
+                    " CONSTRAINT PK PRIMARY KEY (\n" + " TENANT_ID," + " KEY_PREFIX" + ")) MULTI_TENANT=TRUE");
+            stmt.execute("CREATE VIEW " + fullViewName + "(\n" + " ID1 VARCHAR NOT NULL,\n" + " ID2 VARCHAR NOT NULL,\n" + " EVENT_DATE DATE NOT NULL,\n" +
+                    " CONSTRAINT PKVIEW PRIMARY KEY\n" + " (\n" + " ID1, ID2 DESC, EVENT_DATE DESC\n" + ")) " +
+                    "AS SELECT * FROM " + fullTableName + " WHERE KEY_PREFIX = '0CY'");
+        }
+
+
+        try (Connection viewConn = DriverManager.getConnection(TENANT_SPECIFIC_URL1)) {
+            viewConn.setAutoCommit(true);
+            Statement stmt = viewConn.createStatement();
+            stmt.execute("CREATE VIEW IF NOT EXISTS " + tenantView + " AS SELECT * FROM " + fullViewName);
+            viewConn.createStatement().execute("UPSERT INTO " + tenantView + "(ID1, ID2, EVENT_DATE) VALUES " +
+                    "('005xx000001Sv6o', '000000000000300', 1532458230000)");
+            viewConn.createStatement().execute("UPSERT INTO " + tenantView + "(ID1, ID2, EVENT_DATE) VALUES " +
+                    "('005xx000001Sv6o', '000000000000400', 1532458240000)");
+            viewConn.createStatement().execute("UPSERT INTO " + tenantView + "(ID1, ID2, EVENT_DATE) VALUES " +
+                    "('005xx000001Sv6o', '000000000000500', 1532458250000)");
+            viewConn.commit();
+
+            ResultSet rs = stmt.executeQuery("EXPLAIN DELETE FROM " + tenantView + " WHERE (ID1, ID2, EVENT_DATE) " +
+                    "IN (('005xx000001Sv6o', '000000000000400', 1532458240000)," +
+                    "('005xx000001Sv6o', '000000000000300', 1532458230000))");
+            assertTrue(getExplainPlanString(rs).contains("CLIENT PARALLEL 1-WAY RANGE SCAN"));
+
+            rs = stmt.executeQuery("EXPLAIN DELETE FROM " + tenantView + " WHERE (ID1, EVENT_DATE, ID2) IN " +
+                    "(('005xx000001Sv6o', 1532458240000, '000000000000400')," +
+                    "('005xx000001Sv6o', 1532458230000, '000000000000300'))");
+            assertTrue(getExplainPlanString(rs).contains("CLIENT PARALLEL 1-WAY RANGE SCAN"));
+
+            rs = stmt.executeQuery("EXPLAIN DELETE FROM " + tenantView + " WHERE (ID2, EVENT_DATE, ID1) IN " +
+                    "(('000000000000400', 1532458240000, '005xx000001Sv6o')," +
+                    "('000000000000300', 1532458230000, '005xx000001Sv6o'))");
+            assertTrue(getExplainPlanString(rs).contains("CLIENT PARALLEL 1-WAY RANGE SCAN"));
+
+            rs = stmt.executeQuery("EXPLAIN DELETE FROM " + tenantView + " WHERE (EVENT_DATE, ID2, ID1) IN " +
+                    "((1532458240000, '000000000000400', '005xx000001Sv6o')," +
+                    "(1532458230000, '000000000000300', '005xx000001Sv6o'))");
+            assertTrue(getExplainPlanString(rs).contains("CLIENT PARALLEL 1-WAY RANGE SCAN"));
+
+            stmt.execute("DELETE FROM " + tenantView + " WHERE (EVENT_DATE, ID2, ID1) IN " +
```

Comment:
Do we need this delete what is this testing to the this test case?

Reply:
Yes. I do want to check if my changes still delete the right row.